### PR TITLE
Reformat snowDA templates to jinja2

### LINKED ISSUE #2347: Rename config.base.emc.dyn to config.base
### What new functionality do you need?

We no longer carry a `static.nco` version of `config.base` in the repository, so we can rename the `config.base.emc.dyn` to just `config.base`.

### What are the requirements for the new functionality?

`config.base.emc.dyn` renamed to `config.base`.

### Acceptance Criteria

- [ ] `config.base.emc.dyn` renamed to `config.base` for all model systems (`gfs`/`gefs`/`sfs`)
- [ ] Workflow generation updated to use the renamed file as the template for the experiment `config.base`
- [ ] Output identical to existing

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by aerorahul**
# Description
This PR:
- replaces use of non-jinja2 templates in the yaml templates.  Specifically `$( )` in favor of pure jinja2.
- uses jinja2's built-in capability to include templates within templates, thereby allowing to assemble a completely rendered template before passing it to for e.g. yaml loader.
- requires updates to `wxflow` and `gdasapp`
- Changes in `wxflow` in `parse_j2yaml` are **not** backwards compatible 

Additionally, this PR:
- renames `config.base.emc.dyn` to `config.base`.  Resolves #2347 

Depends on PRs in gdasapp and wxflow:
- https://github.com/NOAA-EMC/GDASApp/pull/947
- https://github.com/NOAA-EMC/wxflow/pull/15

# Type of change
- New feature (adds functionality)
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? YES [possibly breaks jedi atm, aero]. And soca if it uses `parse_j2yaml`
- Does this change require a documentation update? NO

# How has this been tested?
- SnowDA CI test on Hera

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by jiaruidong2017 — COMMENTED** [review]
In `parm/config/gfs/config.prepsnowobs`:
Change:
```
export GTS_OBS_LIST="${HOMEgfs}/sorc/gdas.cd/parm/snow/prep/prep_gts.yaml"
export IMS_OBS_LIST="${HOMEgfs}/sorc/gdas.cd/parm/snow/prep/prep_ims.yaml"
```
TO:
```
export GTS_OBS_LIST="${HOMEgfs}/sorc/gdas.cd/parm/snow/prep/prep_gts.yaml.j2"
export IMS_OBS_LIST="${HOMEgfs}/sorc/gdas.cd/parm/snow/prep/prep_ims.yaml.j2"
```

**Comment by aerorahul** [comment]
> In `parm/config/gfs/config.prepsnowobs`: Change:
> 
> ```
> export GTS_OBS_LIST="${HOMEgfs}/sorc/gdas.cd/parm/snow/prep/prep_gts.yaml"
> export IMS_OBS_LIST="${HOMEgfs}/sorc/gdas.cd/parm/snow/prep/prep_ims.yaml"
> ```
> 
> TO:
> 
> ```
> export GTS_OBS_LIST="${HOMEgfs}/sorc/gdas.cd/parm/snow/prep/prep_gts.yaml.j2"
> export IMS_OBS_LIST="${HOMEgfs}/sorc/gdas.cd/parm/snow/prep/prep_ims.yaml.j2"
> ```

Thanks for catching this.
Done in [4528c74](https://github.com/NOAA-EMC/global-workflow/pull/2371/commits/4528c741178a003587545a9235431268a0bb91c5)

**Review by jiaruidong2017 — COMMENTED** [review]
In `parm/config/gfs/config.snowanl`, please make the following changes from the right windows to the left windows as below: 

image

**Comment by aerorahul** [comment]
> In `parm/config/gfs/config.snowanl`, please make the following changes from the right windows to the left windows as below:
> 
> image

Thanks also for catching this.  I had made the change in my experiment directory but had not committed it to HOMEgfs.
I have reworked some of the logic for the obslist.  
A test is running on Hera at: `/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNTESTS/EXPDIR/snowda`
I will post an update when its complete.

**Comment by aerorahul** [comment]
@jiaruidong2017 @CoryMartin-NOAA 
I ran the snowDA CI test overnight and it completed successfully.
The experiment and COM directories can be accessed on Hera at:
`/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNTESTS/EXPDIR/snowda` 
`/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNTESTS/COMROOT/snowda`

**Review by jiaruidong2017 — APPROVED** [review]
Thanks for your efforts.

**Review by CoryMartin-NOAA — APPROVED** [review]
Looks good, thanks @aerorahul !

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Looks good once the dependent PRs are merged and git modules are updated.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by TerrenceMcGuinness-NOAA** [comment]
Jenkins Artifacts captured appropriate[ error log](https://jenkins.epic.oarcloud.noaa.gov/blue/organizations/jenkins/global-workflow%2FEMC-Global-Pipeline/detail/PR-2371/1/artifacts) for an issue with `wxflow/yaml_file.py` in **prepoceanobs**

**Comment by aerorahul** [comment]
The failed test is `C48mx500_3DVarAOWCDA` specifically the `prepoceanobs` task.  
The changes are not likely going to be resolved soon, so we will have to disable this test.

File: ci/cases/pr/C48mx500_3DVarAOWCDA.yaml [hunk 1]
```diff
@@ -19,4 +19,5 @@ arguments:
 
 skip_ci_on_hosts:
   - orion
+  - hera
```
Comment (guillaumevernieres) [inline ci/cases/pr/C48mx500_3DVarAOWCDA.yaml:22]: Do you remember what the issue was @aerorahul ?

**Pull request merged**